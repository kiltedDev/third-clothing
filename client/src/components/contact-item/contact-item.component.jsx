import React from 'react';
import { withRouter } from 'react-router-dom';

import Icon from '../../icons/icon.index.js';

import {
  ContactItemContainer,
  ContactItemIcon,
  ContentContainer,
  ContactTitle,
  ContactSubtitle
} from './contact-item.styles';

const ContactItem = ({ title, contactInfo, iconUrl, linkTo, iconSize }) => (
  <ContactItemContainer
    className='contact-item'
  >
    <ContentContainer className='content'>
      <ContactItemIcon
        href={ linkTo }
        className='contact-icon'
        iconSize={ iconSize }
      >
      <Icon title={ title } />
    </ContactItemIcon>
      <ContactTitle
        className='title'
        children={ title.toUpperCase() }
      />
      <ContactSubtitle
        name={ contactInfo }
        href={ linkTo }
        className='deets'
        children={ contactInfo }
      />
    </ContentContainer>
  </ContactItemContainer>
)

export default withRouter( ContactItem );
