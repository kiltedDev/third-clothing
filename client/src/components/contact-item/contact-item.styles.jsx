import styled, { css } from 'styled-components';

export const ContactItemContainer = styled.div`
	width: 190px;
	display: inline-block;
	align:center;
	float: none;
	vertical-align: top;
`;

export const ContentContainer = styled.div`
  height: 100%;
	width: 100%
  padding: 0 25px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background-color: white;
  opacity: 0.7;
`;

export const ContactItemIcon = styled.a`
	width: 120px;
	height: 120px;
	background-position: center;
	background-repeat: no-repeat;
	background-image: ${({ imageUrl }) => `url(${imageUrl})`};
	background-color: #ccc;
	border: 1px solid #333;
	border-radius: 1000px;
	transition: all .25s;
	cursor: pointer;

	svg {
		position: relative;
		top: 50%;
		left: 50%;
		transform: translate(-50%, -50%);
		width: ${({ iconSize }) => `${iconSize}`};
	}
	&:hover {
		background-color: #777;
	}
`;

export const ContactTitle = styled.span`
	margin-top: 20px;
	font-weight: bold;
  margin-bottom: 6px;
  font-size: 22px;
  color: #4969e1;
`;

export const ContactSubtitle = styled.a`
	margin-top:10px;
  font-weight: lighter;
  font-size: 16px;
	cursor: pointer;
`;
