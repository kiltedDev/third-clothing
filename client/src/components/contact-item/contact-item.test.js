import { shallow } from 'enzyme';
import React from 'react';
import { ContactItem } from './contact-item.component';

describe('ContactItem', () => {
  let wrapper;

  beforeAll(() => {
    const mockProps = {
      title: 'FaceSpace',
      contactInfo: 'cool-guy-82',
      linkTo: 'www.facespace.com/cool-guy-82',
      iconSize: '50px'
    }
    wrapper = shallow(<ContactItem { ...mockProps } />)
  })
  it('should render ContactItem component',() => {
    expect( wrapper.debug() ).toMatchSnapshot();
  })
})
