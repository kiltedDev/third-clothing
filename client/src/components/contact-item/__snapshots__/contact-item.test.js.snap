// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactItem should render ContactItem component 1`] = `
"<styled.div className=\\"contact-item\\">
  <styled.div className=\\"content\\">
    <ContactItemIcon href=\\"www.facespace.com/cool-guy-82\\" className=\\"contact-icon\\" iconSize=\\"50px\\">
      <Icon title=\\"FaceSpace\\" />
    </ContactItemIcon>
    <ContactTitle className=\\"title\\">
      FACESPACE
    </ContactTitle>
    <styled.a name=\\"cool-guy-82\\" href=\\"www.facespace.com/cool-guy-82\\" className=\\"deets\\">
      cool-guy-82
    </styled.a>
  </styled.div>
</styled.div>"
`;
